src/main/java/org/mockito/plugins/MockMaker.java

```diff
@@ -108,6 +109,26 @@
     @Incubating
     TypeMockability isTypeMockable(Class<?> type);
 
+    /**
+     * If you want to provide your own implementation of {@code MockMaker} this method should:
+     * <ul>
+     *     <li>Alter the supplied class to only change its behavior in the current thread.</li>
+     *     <li>Only alters the static method's behavior after being enabled.</li>
+     *     <li>Stops the altered behavior when disabled.</li>
+     * </ul>
+     *
+     * @param settings Mock creation settings like type to mock, extra interfaces and so on.
+     * @param handler See {@link org.mockito.invocation.MockHandler}.
+     *                <b>Do not</b> provide your own implementation at this time. Make sure your implementation of
+     *                {@link #getHandler(Object)} will return this instance.
+     * @param <T> Type of the mock to return, actually the <code>settings.getTypeToMock</code>.
+     * @return A control for the static mock.
+     * @since 3.4.0
+     */
+    @Incubating
+    <T> StaticMockControl<T> createStaticMock(
```

Comment:
This is a breaking change and breaks Dexmaker. I recommend adding a default implementation with ```throw new MockitoException("The subclass byte code generator cannot create static mocks");```

Reply:
You are right, I added a default method which throws an explaining exception. I also had to move the static mock control into a non-internal package. Otherwise nobody else could implement the API.